domain/application/state/application.go

```diff
@@ -3658,8 +3661,7 @@ func decodeConstraints(cons applicationConstraints) constraints.Value {
 
 	// Add the unique spaces, tags and zones to the result:
 	if len(spaces) > 0 {
-		spacesSlice := spaces.SortedValues()
-		res.Spaces = &spacesSlice
+		res.Spaces = ptr(slices.Collect(maps.Values(spaces)))
```

Comment:
I hope the go compiler can see these and fuse these two methods, which ultimately is 2 for loops into one.

Reply:
I think I read somewhere that that's the case. I'll try to find it. In any case, I hope they did plan for this when implementing otherwise it's useless to have iterators.